### PR TITLE
fix(gui): reopen the connection dialog when startup auto-connect gives up — Principle XI.

### PR DESCRIPTION
**Pull request by crypticpy**
Addresses #5367's early directed-probe startup failures: clear the looking-for-radio overlay, surface the reason in the connection dialog, and show the dialog without toggling an already open window closed. Session authentication failure, broadcast-only saved discovery, and stale routed-serial mismatches remain outside this fix.

Startup failure reporting now ends its ownership after a report, a successful probe handoff, connection, or operator route edits. The current-main ANAN probe exits follow the same contract. The HL2 negative test injects a completed result into production handling; it no longer sends a datagram or assumes socket binding succeeded.

Scope: ConnectionPanel.cpp/.h, the MainWindow startup hook, startup_autoconnect_lockout_test.cpp and tests/tests.cmake. No new radio transport, thread, dependency, UI design, workflow or governance change.

Validation at signed head4a5824ee9dc6d6a7e91c70743ce3c71e217165ce:

- Socket-free focused test passed all 18 assertions. Removing HL2 reporting failed three assertions; removing dispatch ownership clearing failed three assertions. Source was restored and rebuilt; all18 passed again.
- Standalone ConnectionPanel fixture owns no discovery/backend, blocks Icom family-hint keychain loading, stages synthetic credentials, bypasses DNS with numeric input, and stops at a signal spy without a transport recipient.
- Full Mac arm64 Release app built and development signing verified; ASR/RADE/DFNR disabled. All five GitHub checks passed, including Mac/Windows/Linux builds and static checks.
- Strict registration, frozen CI gate, engine boundary, generated manifest, diff and ARM RNNoise graph checks passed.

No live radio, RF/TX, real credential-store or native startup-recovery session is claimed. A full native startup would start background discovery and restore Icom keychain state before the credential bail; isolated negative behavior is verified at the socket-free seam instead. Maintainer review clearance remains required.

**Not closed by this PR** (pre-existing, for the record): a broadcast-only saved radio that is powered off never runs a probe, so nothing can give up; a routed probe that answers but is dropped by the `routedRadioFound` serial check (a Flex reached by hostname hits this every launch, since the saved serial is the hostname and the saved routed IP is numeric) reports nothing; and a Flex probe whose TCP connect succeeds but whose peer never sends `H` has no exit at all. These narrow the overlay class rather than close it.

### CODE REVIEW

**Review by ten9876 — CHANGES_REQUESTED** [review]
## Issue fit

#5367: a saved radio unreachable at startup left the operator on a permanent "Looking for your radio…" overlay — the connection-dialog popup is suppressed whenever `LastConnectedRadioSerial` is set, so a bail that only calls `setManualMessage()` writes onto a page behind a dialog that never opens. The mechanism the PR adds — `startupConnectUnavailable` handed up to MainWindow, gated on `m_startupProbe` (latched across the Icom keychain re-entry, cleared on proven connect / manual takeover) — is exactly right, cleanly reasoned, and well-tested (`startup_autoconnect_lockout_test` pins the startup-reports / interactive-stays-silent contract). The MainWindow handler guards (`!m_userDisconnected && !isConnected()`, show-not-toggle) are correct.

But the fix is wired into only some of `probeRadio`'s bail paths, and two of the ones it misses are central cases of "saved radio cannot be reached" — the very symptom.

## Scope

Three files + test, all on-issue. `CHANGELOG.md` untouched. Preflight: no sockets, no fake peers.

## Blockers

**1. Two startup-reachable bails still strand the overlay** (both untouched context, so referenced by line rather than anchored inline). `reportStartupProbeFailure` was added to the source-path-unavailable and Icom-credential bails, but not to:

- **DNS resolution failure** — `src/gui/ConnectionPanel.cpp:2752` (`"Could not resolve \"%1\"…"`): a saved radio addressed by hostname whose name stops resolving (DHCP/DNS change) bails with `setManualMessage` only.
- **HL2 no-answer** — `src/gui/ConnectionPanel.cpp:2795` (`"No Hermes-Lite 2 answered at %1…"`, the `Hl2ProbeResult::NoAnswer` arm): a saved Hermes-Lite 2 powered off or unreachable at startup bails with `setManualMessage` only.

Both are reached on the `restoreSavedFamily` startup path (the saved route resolves to a hostname / to HL2), so an operator with a saved HL2 idle at boot, or a saved `.local`/DNS name that moved, gets the identical stuck overlay #5367 describes. Wrap each in `if (!reportStartupProbeFailure(<short reason>)) { setManualMessage(<full reason>, true); }` before the `return` (for HL2, before the shared `return` covering `NoAnswer`), and add an HL2-no-answer startup row to the test so completeness is pinned rather than spot-fixed. Any other `probeRadio` bail reachable under `restoreSavedFamily` wants the same treatment — worth an audit pass rather than fixing these two by name.

## Nits (non-blocking)

- The three `m_startupProbe = false` sites (proven-connect at :1401, manual-takeover at :2499, bail-consumed in `reportStartupProbeFailure`) are each correct and distinct — a one-line "cleared in three places" note near the member declaration would stop a future reader taking one for redundancy.

## What was verified vs read

- **Verified by me in the PR head:** the three clear-sites are distinct and correct; the Icom-keychain re-entry latch reasoning holds; and — the blocker — the DNS-fail and HL2-no-answer bails call only `setManualMessage` with no upward report, both reachable under `restoreSavedFamily`.
- **Read, not re-run:** the MainWindow handler guards and the test's startup/interactive contract (CI hasn't run — a merge gate).
- The automated pass for this PR misfired (it returned #5364's SpectralNR findings) and was discarded; this is a direct manual pass. No bridge session — the stuck-overlay paths are startup-timing GUI state, and the PR's own extracted test is the right coverage vehicle once the two missing bails are wired in.

**Comment by crypticpy** [comment]
@ten9876 Thanks — both bails wired, and the audit pass you suggested found more of the same shape. Pushed as 58ffe694.

**Bails now reporting upward** (every `probeRadio()` exit reachable under `restoreSavedFamily` that does not end in a connect request):

| Path | Bail | Before |
|---|---|---|
| Icom | hostname does not resolve | stranded |
| HL2 | no answer within 600 ms | stranded |
| HL2 (inside `probeHermesLite2`) | explicit bind failed / could not open UDP socket / name has no IPv4 / datagram send failed / radio already in use by another client | stranded |
| Flex | explicit bind failed / 3 s no-response timeout / socket error (refused, host not found) | stranded |

The Flex three were the surprise: a saved *routed* Flex powered off at boot had the identical stuck overlay #5367 describes, since `LastRoutedRadioIp` goes through `probeFlexRadio()` rather than discovery. The one `probeRadio()` return I left alone is the unparseable custom CI-V address — `IcomSettings` only persists validated bytes, so a startup probe cannot reach it.

**Two small cleanups of my own** while in there: `reportStartupProbeFailure()` moves to the private section (nothing outside `ConnectionPanel` called it) and drops its unused `bool` return. I kept the full `setManualMessage()` text on every bail rather than the `if (!report…)` form — once the dialog opens the operator lands on the manual page, and the longer message with the "check the radio is powered, idle…" guidance is the useful one there; the short reason goes to the status line.

**Test:** `startup_autoconnect_lockout_test` gains the HL2 no-answer row. It lets the real probe run against 192.0.2.10 (TEST-NET-1) and waits out its own 600 ms deadline; dropped, refused by the local stack, or unable to bind are all startup bails, so it holds regardless of the runner's network. Disclosed in the `tests.cmake` block per the AGENTS.md carve-out: one ephemeral UDP port, one datagram, no peer. Mutation-checked — removing only the `NoAnswer` report fails both new assertions. I did not add DNS or Flex rows: those need a resolver or a TCP peer in the test, which the test-layer boundary rules out, and the wiring is the same one-liner pattern as the HL2 arm.

**Nit:** the "cleared in three places, none redundant" note is on the `m_startupProbe` declaration.

**Review by Ozy311 — COMMENTED** [review]
Re-reviewed and repaired exact head4a5824ee9dc6d6a7e91c70743ce3c71e217165ce. Both cited blockers are removed: the negative HL2 row now injects the production result without a socket, and the address assertion only tests that injected NoAnswer outcome. Startup ownership also clears at probe dispatch and manual route edits; current-main ANAN bails are preserved and reported consistently. The header/body scope excludes session authentication and broadcast-only recovery.

Five source/test paths resolve to Tier3 reviewers CODEOWNERS. Signed commit verified; all five current PR checks passed. The standalone test passed18 assertions, both deliberate mutations failed three assertions each, and the restored source was rebuilt and passed. Full signed Mac app build passed. No native startup/keychain or live radio/RF/TX result is claimed; that path would invoke real discovery and keychain work, so its negative contract remains at the isolated seam.

No remaining code blocker found in this review. This COMMENT supplies the completed repair evidence; it does not dismiss or override ten9876's existing change-request decision. Current main contains release metadata and the separate Ulanzi handler merge; the startup/test changes do not overlap those changes.

**Comment by ten9876** [comment]
@crypticpy thanks for all of the work you've done submitting PRs for the project.  We would love to get you more directly involved with the core dev team.  If possible, please join our discord server, we'd love to have you there.

Cheers,

https://discord.gg/kpsZctwb

File: src/gui/MainWindow.cpp [hunk 1]
```diff
@@ -2578,6 +2578,23 @@ MainWindow::MainWindow(QWidget* parent)
...
     });
 
+    // A startup auto-connect that gives up must hand the window back. The
+    // no-saved-radio popup above is suppressed whenever LastConnectedRadioSerial
+    // is set, so without this a saved radio that cannot be reached leaves the
+    // "Looking for your radio…" overlay up forever with no offered way into the
+    // connection dialog — the operator has to know the Settings menu item exists.
+    connect(m_connPanel, &ConnectionPanel::startupConnectUnavailable,
+            this, [this](const QString& reason) {
+        if (m_userDisconnected || m_radioModel.isConnected()) {
```
Comment (ten9876) [inline src/gui/MainWindow.cpp]: **Blocker 2 — guard misses a connect that is already in flight.** `isConnected()` is false for the whole handshake (`RadioModel.h:175-183`, which exists precisely so callers can tell "working" from "idle"), and `maybeAutoConnectToDiscoveredRadio()` guards its own self-collision at `MainWindow_Session.cpp:527` for the same reason. With an HL2/ANAN last connected by IP on the LAN, discovery starts the auto-connect at ~100 ms and the 500 ms routed probe then sees the radio busy with our own stream → "in use by another client" → this handler tears down the live connecting overlay (and the #5052 WDSP dialog), overwrites the footer, and pops the dialog over a connect that succeeds moments later. Silent on `main`; a regression here.

```suggestion
        if (m_userDisconnected || m_radioModel.isConnected()
            || m_radioModel.isConnectAttemptInFlight()) {
```

(Reasoned from code — the demo radio does not go through discovery auto-connect, so I could not stage the race offscreen. If you can show the directed HL2 probe never reports busy during our own handshake, I will withdraw it.)

File: tests/tests.cmake [hunk 2]
```diff
@@ -4482,6 +4502,7 @@ target_link_libraries(CAT_Flex_test PRIVATE Qt6::Core Qt6::Network)
 # directly (rather than linking aethercore) needs the vendored SQLite engine.
 # Conditional targets are guarded with if(TARGET ...).
 set(AETHER_SETTINGS_CONSUMERS
+    startup_autoconnect_lockout_test
```
Comment (ten9876) [inline tests/tests.cmake]: Nit, and the merge-conflict site. The list's header (four lines up) scopes it to targets that compile `${AETHER_SETTINGS_SOURCES}` *directly rather than linking aethercore*; this target links `aethercore`, and its twin `connection_panel_size_test` (same three sources, same link line) is not listed. `AGENTS.md`'s broader "touches AppSettings" wording contradicts the comment — that is a docs defect, not yours — but dropping the entry removes both the redundant `aether_sqlite3` link and the conflict with `main`.

```suggestion
```

File: src/gui/ConnectionPanel.cpp [hunk 3]
```diff
@@ -3199,6 +3280,8 @@ ConnectionPanel::AnanProbeResult ConnectionPanel::probeAnan(
                                "ANAN-G2: %1").arg(hpsdr.errorString()),
                 true);
         }
+        reportStartupProbeFailure(
+            QStringLiteral("Could not probe the ANAN-G2 at %1. Check the source path and address.").arg(ip));
```
Comment (ten9876) [inline src/gui/ConnectionPanel.cpp]: Nit: this generic reason is used for three different faults here and at `:3323` (explicit-bind failure, socket open failure, datagram send failure), while `probeHermesLite2` keeps them distinct (`:3103` / `:3110` / `:3166`). The two families now diverge in what the footer says for the same fault — consider mirroring the HL2 wording, or (simpler) passing the text `setManualMessage()` already built instead of a hand-shortened second copy.

File: src/gui/ConnectionPanel.cpp [hunk 4]
```diff
@@ -2774,6 +2815,7 @@ void ConnectionPanel::probeRadio(const QString& ip, bool restoreSavedFamily)
                     || panel->m_manualIpEdit->text().trimmed() != requestedHost) {
                     panel->resetManualConnectButton();
                     panel->m_manualConnectPending = false;
+                    panel->m_startupProbe = false;
```
Comment (ten9876) [inline src/gui/ConnectionPanel.cpp]: Nit: this drops the latch silently. As far as I can trace, every way `currentManualFamily()` or the host text can change while the read is pending is an operator gesture that already cleared the latch (`textEdited`, combo `activated`, manual connect), so this line is either redundant or — if some programmatic path can change the family under a pending read — a stuck-overlay path with no report. Either report upward here too or drop the assignment and let the invariant carry it; the `probeRadio` comment's list of clearing sites would then stay accurate.

File: src/gui/ConnectionPanel.cpp [hunk 5]
```diff
@@ -2848,6 +2892,7 @@ void ConnectionPanel::probeRadio(const QString& ip, bool restoreSavedFamily)
                                           .arg(m_manualIcomCivEdit->text().trimmed()));
                         m_manualIcomCivEdit->setFocus();
                         m_manualIcomCivEdit->selectAll();
+                        reportStartupProbeFailure(tr("The saved Icom CI-V address is invalid."));
```
Comment (ten9876) [inline src/gui/ConnectionPanel.cpp]: Nit: the only `tr()` among the new reasons (siblings are all `QStringLiteral`). Also, because this reports upward, MainWindow's `setStatusText(reason)` immediately replaces the more specific `CI-V address "…" is not a hex byte…` footer written two lines above. Unreachable from startup today (persisted CI-V values are validated), so cosmetic.

File: src/gui/ConnectionPanel.cpp [hunk 6]
```diff
@@ -3489,6 +3582,8 @@ void ConnectionPanel::probeFlexRadio(const QString& trimmedIp, const RadioBindSe
         m_manualConnectPending = false;
         m_manualConnectBtn->setText("Connect by IP");
         updateActionState();
+        reportStartupProbeFailure(
+            QStringLiteral("Could not reach %1: %2").arg(trimmedIp, sock->errorString()));
```
Comment (ten9876) [inline src/gui/ConnectionPanel.cpp]: Nit: same string formatted twice (`:3579` and here), including two `sock->errorString()` calls. Build it once into a local and pass it to both.

File: tests/startup_autoconnect_lockout_test.cpp [hunk 7]
```diff
@@ -0,0 +1,286 @@
...
+    auto* host = panel.findChild<QLineEdit*>(QStringLiteral("connectionManualIp"));
+    auto* family = panel.findChild<QComboBox*>(QStringLiteral("connectionManualRadioType"));
+    report("operator route controls exist", host && family);
+    if (!host || !family) {
+        return;
+    }
+    ConnectionPanelStartupTestAccess::arm(panel);
+    // textEdited/activated are the user-only signals; startup's setText and
+    // saved-family restoration must not cancel the startup probe themselves.
+    QMetaObject::invokeMethod(host, "textEdited", Qt::DirectConnection,
```
Comment (ten9876) [inline tests/startup_autoconnect_lockout_test.cpp]: Nit: `invokeMethod` by signal *name* is a string lookup whose `bool` return is unchecked — a rename compiles and only shows up as this row failing with no hint why. Qt signals are public, so `emit host->textEdited(QStringLiteral("replacement-host"));` (and `emit family->activated(0);` below) is compile-checked and shorter.

File: src/gui/ConnectionPanel.cpp [hunk 8]
```diff
@@ -2970,6 +3010,36 @@ void ConnectionPanel::probeRadio(const QString& ip, bool restoreSavedFamily)
...
+        reportStartupProbeFailure(
+            QStringLiteral("No Hermes-Lite 2 answered at %1.").arg(trimmedIp));
+    } else if (probe == Hl2ProbeResult::NotAttempted) {
+        // probeHermesLite2() has already reported its own reason, upward
+        // included; only the button needs restoring here.
+        resetManualConnectButton();
+    }
+}
+
+void ConnectionPanel::finishManualProbe(const RadioInfo& info, bool routedOnly)
```
Comment (ten9876) [inline src/gui/ConnectionPanel.cpp:3042]: Note (not a request): clearing the latch on dispatch is right — it is what stops a later session-level failure from popping the dialog with a stale reason. The flip side is that a routed probe that *succeeds* but is then dropped by MainWindow's `routedRadioFound` serial check (`MainWindow.cpp:2565`) can no longer be reported, and the overlay stays. A routed Flex reached by **hostname** hits that on every launch (`info.serial` = hostname, `LastRoutedRadioIp` = numeric peer address). Pre-existing and conceded in the body; a line naming it there would keep the next reader honest.

**Review by ten9876 — DISMISSED** [review]
## Issue fit

#5367: a saved radio that cannot be reached at startup leaves the operator on a permanent "Looking for your radio…" overlay, because the no-saved-radio dialog popup is suppressed whenever `LastConnectedRadioSerial` is set and every `probeRadio()` bail only writes to the hidden manual page. The PR's mechanism — a `m_startupProbe` latch set only by the startup `restoreSavedFamily` call, `reportStartupProbeFailure()` emitting `startupConnectUnavailable(reason)` from every startup-reachable bail, and a MainWindow handler that clears the overlay, writes the reason to the footer and *shows* (not toggles) the dialog — is the right shape, and it now covers all three bails the issue names plus the HL2/ANAN/Flex directed-probe exits. **Reproduced before/after against a built merge of this head onto current `main`** (details under "Verified"): on `main` the stale-route launch ends with `ConnectionPanel visible=False` and no reason anywhere; on this head the dialog is up with the footer reading `Could not reach 127.0.0.1: Connection refused` (Flex) / `This Icom needs its network user name and password.` (Icom, no username). The PR's own 18-assertion test passes and the three mutations I ran (HL2 no-answer report removed, `finishManualProbe` latch clear removed, `emit` removed) each fail it, matching the body's claims.

Two things stop it merging as-is: the branch **conflicts with `main`** (`tests/tests.cmake`), and the new MainWindow guard **does not cover a connect that is already in flight**, which turns a previously-silent stale-route failure into a dialog popping over a succeeding auto-connect (blocker 2, reasoned from code — see the scenario; I could not reach it with the demo radio, and would welcome a refutation).

## Scope

| File / group | What it changes | Claimed in title/body? | Verdict |
|---|---|---|---|
| `src/gui/ConnectionPanel.h` | `startupConnectUnavailable` signal, `reportStartupProbeFailure` / `handleHl2ProbeResult` / `finishManualProbe` decls, `m_startupProbe`, test-access friend | Yes | In scope |
| `src/gui/ConnectionPanel.cpp` | latch set/clear sites, ~20 `reportStartupProbeFailure` calls across `probeRadio` / `probeHermesLite2` / `probeAnan` / `probeFlexRadio`, HL2 result handling extracted, five `emit connectRequested`/`routedRadioFound` routed through `finishManualProbe` | Yes | In scope; the `-` lines are a pure move (HL2 block → `handleHl2ProbeResult`, identical text/order), no guard removed |
| `src/gui/MainWindow.cpp` | one `connect()` consuming the signal | Yes | In scope |
| `tests/startup_autoconnect_lockout_test.cpp` | new socket-free harness (18 assertions) | Yes | In scope |
| `tests/tests.cmake` | registers the target; adds it to `AETHER_SETTINGS_CONSUMERS` | Yes | In scope; the consumers-list entry is redundant and is the merge-conflict site (nit below) |

Everything in the diff is explained by the issue. No `CHANGELOG.md` entry (correct), no CI edits, no formatting churn, no new protocol/config/CLI surface — the only new public surface is one Qt signal on an internal GUI class. Commit `58ffe694`'s message still describes the UDP-owning HL2 test row that `4a5824ee` replaced; whoever writes the squash message should not carry that paragraph forward.

**Test-boundary preflight:** the harness at this head is socket-free. `probeRadio` is only entered with the Icom family selected (credential bail before any transport, or numeric-address dispatch that stops at `connectRequested` with no recipient), the HL2 no-answer row injects `Hl2ProbeResult::NoAnswer` into `handleHl2ProbeResult`, and no `QUdpSocket`/`QTcpSocket`/`listen`/`bind`/`connectToHost` is reached. It is not in any `ci.yml` `-R` filter, so the green checks did not run it; I ran it locally.

## Blockers

1. **Branch conflicts with `main`** (`mergeable: CONFLICTING`). `origin/main` gained four entries at the same spot in `AETHER_SETTINGS_CONSUMERS` (`client_display_settings_test`, `rx_applet_squelch_reconciliation_test`, `rtl_slice_settings_test`, `weather_radar_loading_test`). Rebase or merge; the current CI green is from 2026-09-06 and did not test the merge with today's `main`. I resolved it locally (keep both) and the merged tree builds and passes the new test, so nothing beyond the conflict is waiting. Dropping the entry (see the inline nit on `tests.cmake:4505`) resolves it by construction.

2. **The MainWindow handler fires over a connect that is already in flight** (inline on `src/gui/MainWindow.cpp:2588`). The guard is `m_userDisconnected || m_radioModel.isConnected()`, but `isConnected()` is false for the whole handshake — `RadioModel.h:175-183` says exactly this and provides `isConnectAttemptInFlight()` for it, and `maybeAutoConnectToDiscoveredRadio()` (`MainWindow_Session.cpp:527`) already guards its own self-collision for the same reason. Scenario: an HL2 (or ANAN) last connected **by IP on the LAN**, so both `LastConnectedRadioSerial` (the MAC-derived serial) and `LastRoutedRadioIp` are saved. Next launch, broadcast discovery finds it within ~100 ms and starts the auto-connect; at 500 ms the routed probe hits the same radio, which now answers the directed discovery with 0x03 (our own stream — the comment at `MainWindow_Session.cpp:519-527` describes this) → `probeHermesLite2` reports "already in use by another client" → `reportStartupProbeFailure` → this handler runs with `isConnected()==false`: it clears the "Connecting to radio…" animation (and, via `setPanadapterConnectionAnimation(false)`, dismisses the #5052 WDSP setup dialog), overwrites the footer with "The Hermes-Lite 2 at X is in use by another client.", and pops the connection dialog over a connect that then succeeds. On `main` that probe result was silent, so this is a regression on a plausible configuration. Same shape if the operator opens the dialog and clicks a discovered radio while the 3 s Flex startup probe is still pending (the list connect does not go through `finishManualProbe`, and `setConnected(true)` only lands after the session). Reasoned from code, not reproduced: the demo radio does not enter the discovery auto-connect path, so I could not stage the race offscreen. Fix is one clause — suggestion inline.

## Nits (non-blocking)

- `tests/tests.cmake:4505` — the `AETHER_SETTINGS_CONSUMERS` entry is redundant: the list's own header scopes it to targets that compile `${AETHER_SETTINGS_SOURCES}` directly *rather than* linking `aethercore`, and this target links `aethercore` (its twin `connection_panel_size_test`, same sources, is not listed). `AGENTS.md`'s "touches AppSettings → add it to the list" wording contradicts the list comment, which is a docs defect, not yours — but dropping the line also makes the conflict go away.
- **The dialog opens on the "On This Network" page** (observed, screenshot below), while the fuller guidance written by `setManualMessage()` sits on the hidden "Connect by IP" page. The footer reason is enough to unblock the operator, so this is a polish call: consider `setCurrentMode(ManualMode)` from the handler (or in `reportStartupProbeFailure`) so the reason and the page it explains are the same page.
- `src/gui/ConnectionPanel.cpp:3284/3323` — the two ANAN `NotAttempted` reasons are one generic string for three different faults (explicit-bind failure, socket open failure, datagram send failure), where the HL2 twins at `:3103/:3110/:3166` keep the distinction. `probeAnan` and `probeHermesLite2` now diverge in what they report for the same fault.
- `src/gui/ConnectionPanel.cpp:2818` — the keychain-callback "route changed under us" branch drops the latch silently. Reachable only while the async read is pending and only after an operator gesture that already cleared the latch, so probably harmless; but if it *is* reachable, the overlay is left with no owner. Either report upward there too, or drop the assignment and let the invariant carry it.
- `src/gui/ConnectionPanel.cpp:2895` — the only `tr()` in the PR (every sibling reason is `QStringLiteral`); and because this bail reports upward, MainWindow's `setStatusText(reason)` immediately overwrites the more specific `CI-V address "…" is not a hex byte` footer the line above just wrote. Unreachable from startup today (persisted CI-V values are validated), so cosmetic.
- `src/gui/ConnectionPanel.cpp:3586` — `"Could not reach %1: %2"` is formatted twice verbatim; build it once.
- `tests/startup_autoconnect_lockout_test.cpp:179/184` — `QMetaObject::invokeMethod(host, "textEdited", …)` is a string lookup with an unchecked return; `emit host->textEdited(...)` / `emit family->activated(0)` are compile-checked and one line each.
- Residual permanent-overlay paths this PR does not reach, for the record (all pre-existing, the body already concedes two of them): (a) broadcast-only saved radio powered off — no probe ever runs; (b) `routedRadioFound` serial mismatch (`MainWindow.cpp:2565`) — note that a routed Flex reached by **hostname** hits this on every launch, since `info.serial` is the hostname but `LastRoutedRadioIp` is the numeric peer address; (c) a Flex probe whose TCP connect succeeds but the peer never sends `H` (`ConnectionPanel.cpp:3415` only bails when not `ConnectedState`) — no exit at all, latch stays set. Worth one line in the body so the PR reads as narrowing the class rather than closing it.

## Verified vs. read

**Verified (built and run):** merged `4a5824ee` onto today's `origin/main` (conflict resolved keep-both), full `AetherSDR` + `startup_autoconnect_lockout_test` build clean; test passes 18/18; mutation 1 (HL2 no-answer report removed) → 3 failures; mutation 2 (`finishManualProbe` latch clear removed) → 3 failures; mutation 3 (`emit` removed) → 6 failures; restored → 18/18. Bridge sessions, offscreen, isolated `AETHER_SETTINGS_DIR`, explicit socket, `AETHER_AUTOMATION_NO_TX=1`, no radio ever connected: seeded `LastConnectedRadioSerial`/`LastRoutedRadioIp`=`127.0.0.1` with a `flex` routed profile → on `main` (`0d2f67c1` build) `dumpTree` reports the ConnectionPanel `visible=False` after 12 s; on this head `visible=True` with footer `Could not reach 127.0.0.1: Connection refused`. Seeded an `icom` profile with no username → head shows the dialog with `This Icom needs its network user name and password.`. The panadapter overlay itself is not observable offscreen (no spectrum renderer), so "overlay cleared" is inferred from the handler path, not pixels.

**Read only:** blocker 2 (the in-flight-connect race) and the residual paths; the macOS keychain re-entry (no keychain here).

**Tried and held:** the `startupConnectUnavailable` declaration is genuinely under `signals:` (the hunk header is a git artifact); `probeRadio`'s `setText` fires `textChanged` not `textEdited`, so startup does not cancel itself; the Flex 3 s timer and `errorOccurred` cannot double-report (latch clears before emit); bridge `connect ip` goes through `onManualConnectClicked()` which clears the latch, so a stale latch cannot pop a dialog mid-automation; `QPointer` null-check protects the keychain callback; the friend/test-access resolves in namespace `AetherSDR`.

File: src/gui/MainWindow.cpp (hunk 1 above)
Comment (ten9876) [reply src/gui/MainWindow.cpp]: Fixed in c736e14b: the guard now also returns on `m_radioModel.isConnectAttemptInFlight()`.

File: tests/tests.cmake (hunk 2 above)
Comment (ten9876) [reply tests/tests.cmake]: Dropped in the merge commit be228ee0; that also resolved the conflict with main.

File: src/gui/ConnectionPanel.cpp (hunk 3 above)
Comment (ten9876) [reply src/gui/ConnectionPanel.cpp]: Fixed in c736e14b: ANAN now reports the same three distinct reasons as HL2 (saved source path unavailable / could not open a UDP socket / could not send a discovery request to <dest>).

File: src/gui/ConnectionPanel.cpp (hunk 4 above)
Comment (ten9876) [reply src/gui/ConnectionPanel.cpp]: Fixed in c736e14b: the branch now calls `reportStartupProbeFailure()` (a no-op when an operator edit already cleared the latch) instead of dropping the latch silently.

File: src/gui/ConnectionPanel.cpp (hunk 5 above)
Comment (ten9876) [reply src/gui/ConnectionPanel.cpp]: Fixed in c736e14b: the message is built once as `QStringLiteral` and the same text goes to the footer and upward, so nothing overwrites it.

File: src/gui/ConnectionPanel.cpp (hunk 6 above)
Comment (ten9876) [reply src/gui/ConnectionPanel.cpp]: Fixed in c736e14b: built once into `reason`.

File: tests/startup_autoconnect_lockout_test.cpp (hunk 7 above)
Comment (ten9876) [reply tests/startup_autoconnect_lockout_test.cpp]: Fixed in c736e14b: both rows now `emit` the signal directly.

File: src/gui/ConnectionPanel.cpp (hunk 8 above)
Comment (ten9876) [reply src/gui/ConnectionPanel.cpp:3042]: Noted in the PR body ("Not closed by this PR"), alongside the broadcast-only and silent-TCP-peer paths. Also, c736e14b now opens the dialog on the Connect by IP page so the full guidance and the footer reason are on the same page.

**Review by ten9876 — APPROVED** [review]
Re-reviewed at c736e14b (merge with main + follow-ups). Both blockers and every nit from the earlier pass are addressed: the startup-bail handler now also guards on an in-flight connect attempt, the redundant consumers-list entry is gone (which also cleared the conflict), the dialog lands on the Connect by IP page, and the ANAN/CI-V/Flex/keychain reasons are consistent. Verified locally: startup_autoconnect_lockout_test 18/18 and connection_panel_size_test ALL PASS on the merged tree; a stale-route bridge launch (routed Flex at 127.0.0.1, offscreen, isolated settings, demo path only) opens the dialog with the reason instead of leaving the overlay up. Approving; auto-merge armed pending CI.
